pkg/remote/oci/resolver.go

```diff
@@ -29,12 +33,22 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
-var _ remote.Resolver = (*Resolver)(nil)
+const (
+	KindAnnotation       = "dev.tekton.image.kind"
+	APIVersionAnnotation = "dev.tekton.image.apiVersion"
+	TitleAnnotation      = "dev.tekton.image.name"
+)
 
 // Resolver implements the Resolver interface using OCI images.
 type Resolver struct {
 	imageReference string
 	keychain       authn.Keychain
+	timeout        time.Duration
+}
+
+// NewResolver is a convenience function to return a new OCI resolver instance as a remote.Resolver.
+func NewResolver(ref string, keychain authn.Keychain) remote.Resolver {
+	return &Resolver{imageReference: ref, keychain: keychain, timeout: time.Second * 120}
```

Comment:
Do we want to make the timeout configurable by the user ?
Otherwise, why is it a field in `Resolver` ? :upside_down_face:

Reply:
I figured we would make it a global user preference eventually but I wanted to do that in a follow-up PR.